File: problems/645-Set-Mismatch/solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int duplicate = 0, loss = 0;
        int expectedSum = (1 + nums.length) * nums.length / 2;
        int targetSum = 0;

        HashSet<Integer> set = new HashSet<>();
        for (int n : nums) {
            if (set.contains(n)) {
                duplicate = n;
            } else {
                set.add(n);
            }

            targetSum += n;
        }

        loss = expectedSum - targetSum + duplicate;

        return new int[]{duplicate, loss};
    }
}

File: problems/645-Set-Mismatch/solution.go
func findErrorNums(nums []int) []int {
    sort.Slice(nums, func(a, b int) bool {
        return nums[a] < nums[b]
    })

    duplicate, missing := 0, 0
    for i := 1; i < len(nums); i++ {
        if nums[i] == nums[i-1] {
            duplicate = nums[i]
        } else if nums[i] == nums[i-1]+2 {
            missing = nums[i] - 1
        } else {
            continue
        }
    } 

    if nums[0] != 1 {
        return []int{duplicate, 1}
    } else if nums[len(nums)-1] != len(nums) {
        return []int{duplicate, len(nums)}
    } else {
        return []int{duplicate, missing}
    }
}

